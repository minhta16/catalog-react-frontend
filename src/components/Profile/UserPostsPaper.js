import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Typography, Divider, Paper, Avatar, Grid, IconButton } from '@material-ui/core';
import LibraryAddIcon from '@material-ui/icons/LibraryAdd';
import PostList from './PostList';

export function UserPostsPaper(props) {
  const { username, posts } = props;
  return (
    <Paper className="left-margin">
      <Grid container alignItems="center" justify="space-between">
        <Grid item>
          <Grid container alignItems="center">
            <Avatar
              style={{
                backgroundColor: '#fdb600',
                color: '#fff',
                margin: '0.5rem',
              }}
            >
              {username.charAt(0)}
            </Avatar>
            <Typography variant="body1">{username}</Typography>
          </Grid>
        </Grid>
        <Grid item>
          <IconButton color="primary" component={Link} exact="true" to="/new-item">
            <LibraryAddIcon className="icon-padding" />
            <Typography variant="button">New item</Typography>
          </IconButton>
        </Grid>
      </Grid>
      <Divider />
      <PostList posts={posts} />
    </Paper>
  );
}

UserPostsPaper.propTypes = {
  username: PropTypes.string.isRequired,
  posts: PropTypes.array,
};

UserPostsPaper.defaultProps = {
  posts: [],
};

export default UserPostsPaper;
