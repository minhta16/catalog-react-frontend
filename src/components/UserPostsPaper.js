import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Divider, Paper, Avatar, Grid } from '@material-ui/core';

export function UserPostsPaper(props) {
  const { username, token } = props;
  return (
    <Paper className="left-margin">
      <Grid container alignItems="center">
        <Avatar
          style={{
            backgroundColor: '#fdb600',
            color: '#fff',
            margin: '0.5rem',
          }}
        >
          {username.charAt(0)}
        </Avatar>
        <Typography variant="body1">{username}</Typography>
      </Grid>
      <Divider />
      <Typography variant="body1">{token}</Typography>
    </Paper>
  );
}

UserPostsPaper.propTypes = {
  username: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired,
};

export default UserPostsPaper;
