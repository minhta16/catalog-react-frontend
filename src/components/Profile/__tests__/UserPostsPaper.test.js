/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { UserPostsPaper } from '../UserPostsPaper';

describe('components/PostItem', () => {
  let wrapper;
  let props;
  // let warning;

  // const update = () => {
  //   wrapper.update();
  //   warning = wrapper.find(Warning);
  // };

  const setup = () => {
    wrapper = shallow(<UserPostsPaper {...props} />);
  };

  beforeEach(() => {
    props = {
      username: 'minh',
      posts: [{ name: 'meo', description: 'meo' }],
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
});
