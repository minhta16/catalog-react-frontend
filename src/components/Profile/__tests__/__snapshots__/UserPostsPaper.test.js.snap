// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostItem should render correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="left-margin"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
  >
    <WithStyles(ForwardRef(Avatar))
      style={
        Object {
          "backgroundColor": "#fdb600",
          "color": "#fff",
          "margin": "0.5rem",
        }
      }
    >
      m
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      minh
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Divider)) />
  <PostList
    posts={
      Array [
        Object {
          "description": "meo",
          "name": "meo",
        },
      ]
    }
  />
</WithStyles(ForwardRef(Paper))>
`;
